allure-testng/src/test/java/io/qameta/allure/testng/AllureTestNgTest.java

```diff
@@ -1090,6 +1090,20 @@ public void shouldOverrideParameters() {
                 );
     }
 
+    @AllureFeatures.Attachments
+    @Test
+    public void shouldProduceAttachmentFromWithinTheDataProvider() {
+        final AllureResults results = runTestNgSuites("suites/attachment-from-data-provider.xml");
+
+        assertThat(results.getTestResultContainers())
+                .flatExtracting(TestResultContainer::getBefores)
+                .hasSize(1)
+                .flatExtracting(FixtureResult::getAttachments)
+                .hasSize(1)
+                .element(0)
+                .hasFieldOrPropertyWithValue("name", "String attachment");
```

Comment:
You may also want to verify its type and content. Moreover, I see there were added 2 hooks. Please add some tests for them. Well, any newly added code should be tested.

Reply:
I will try to add something in the evening.

Reply:
The type and content verification was added